src/Connection/Http2Connection.php

```diff
@@ -1467,9 +1478,58 @@ static function () {
 
                             $this->releaseStream($id); // Response has no body, release stream immediately.
                         } else {
-                            $this->trailerDeferreds[$id] = new Deferred;
                             $this->bodyEmitters[$id] = new Emitter;
 
+                            if ($status >= Status::OK && $status < Status::MULTIPLE_CHOICES && $stream->request->getMethod() === 'CONNECT') {
```

Comment:
I'd use the literal numbers here, as that's more useful for ranges. The names are mostly useful for single status codes, but not ranges.